day8/alienalphabet.py

```diff
@@ -0,0 +1,131 @@
+##############
+## Solution ##
+##############
+
+def alienAlphabet(orderedWords):
+    # edge cases
+    if len(orderedWords) <= 0:
+        return ''
+    if len(orderedWords) == 1:
+        return removeDups(orderedWords[0])
+
+    # linearization of a DAG
+    graph = createDependencyGraph(orderedWords)
+    alphabet = topSort(graph)
+
+    return "".join(alphabet)
+
+#############
+## Helpers ##
+#############
+
+def createDependencyGraph(orderedWords):
+    graph = {}
+    for i, word in enumerate(orderedWords[:-1]):
+        nextWord = orderedWords[i+1]
+        shorter = min(len(word), len(nextWord))
+        for j in xrange(shorter):
+            letter = word[j]
+            otherLetter = nextWord[j]
```

Comment:
oh if you were gonna iterate through two things at the same time, it sounds like a perfect opportunity again for zip!

    for letterFrom1stWord, letterFrom2ndWord in zip(word, nextWord):

the variable names i suggested leave something to be desired but you get the idea. `zip` automatically stops at the length of the shortest iterable, but you could use `itertools.zip_longest` if you wanted to keep going in other cases